### PR TITLE
Abyss Integration

### PR DESCRIPTION
**Pull request by dholms**
Expands the labeler abstraction to serve as an automoderator generally & introduces the ability to takedown records through abyss

### CODE REVIEW

File: packages/bsky/src/auto-moderator/abyss.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  async makeReq(did: string, cid: CID): Promise<ScannerResp> {
+    const { stream, contentType } = await resolveBlob(
+      did,
+      cid,
+      this.ctx.db,
+      this.ctx.idResolver,
+    )
+    const { data } = await axios.post(this.getReqUrl({ did }), stream, {
```
Comment (bnewbold) [inline packages/bsky/src/auto-moderator/abyss.ts:47]: Things we might want to do at this HTTP request call site:

- log response status code and duration (ms). we're also doing logging+metrics in abyss, but good to do it on this side of the loadbalancer as well
- have a timeout. 10sec?

**Review by bnewbold — APPROVED** [review]
One small note but looks good otherwise

File: packages/bsky/src/auto-moderator/abyss.ts (hunk 1 above)
Comment (dholms) [reply packages/bsky/src/auto-moderator/abyss.ts:47]: yup both seem reasonable 👍

File: packages/bsky/src/indexer/config.ts [hunk 2]
```diff
@@ -86,6 +92,9 @@ export class IndexerConfig {
     assert(dbPostgresUrl)
     assert(redisHost || (redisSentinelName && redisSentinelHosts?.length))
     assert(indexerPartitionIds.length > 0)
+    assert(imgUriEndpoint)
```
Comment (dholms) [inline packages/bsky/src/indexer/config.ts:95]: btw i think i'm gonna make these not required & if they're not setup, then just no abyss integration

File: packages/bsky/src/auto-moderator/index.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+            'failed to check img for takedown',
+          )
+        }),
+      ])
+    })
+  }
+
+  async labelRecord(uri: AtUri, recordCid: CID, text: string[], imgs: CID[]) {
+    if (uri.collection === ids.AppBskyActorProfile) {
+      // @TODO label profiles
```
Comment (devinivy) [inline packages/bsky/src/auto-moderator/index.ts:103]: This all good to leave as a `@TODO`?

File: packages/bsky/src/auto-moderator/index.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        subject: { uri, cid: recordCid },
+        subjectBlobCids: takedownCids,
+        reason: `automated takedown for labels: ${results.flat().join(', ')}`,
+        createdBy: this.ctx.cfg.labelerDid,
+      })
+      await modSrvc.takedownRecord({
+        takedownId: action.id,
+        uri: uri,
+        blobCids: takedownCids,
+      })
```
Comment (devinivy) [inline packages/bsky/src/auto-moderator/index.ts:150]: For this to be synced with the PDS we want to actually hit the PDS and let it propagate to the appview—a similar approach can be found in `packages/bsky/src/db/periodic-moderation-action-reversal.ts`.

**Review by devinivy — APPROVED** [review]
Left one note re: syncing takedowns which may be important, but generally everything here looks solid!

File: packages/bsky/src/auto-moderator/index.ts (hunk 3 above)
Comment (dholms) [reply packages/bsky/src/auto-moderator/index.ts:103]: yup 👍 will work by just deleting that check. just need client signoff that there won't be adverse side effects

File: packages/bsky/src/auto-moderator/index.ts (hunk 4 above)
Comment (dholms) [reply packages/bsky/src/auto-moderator/index.ts:150]: ah good looks 👍

**Pull request merged**